### DISCUSSION TITLE
Container gets reconstructed very frequent.

### DISCUSSION THREAD

**Discussion by haringsrob** [phpactor/phpactor#2535] [General]
Hey!

Was experimenting a bit again with my laravel integration and trying to do some performance improvements.

I have a class which is registered in the container which holds a cache of all the properties, views etc.

However, I notice that this is getting re-constructed basically every keystroke. 

Is this intentional or am I doing something wrong?

The LaravelContainerInspector holds the cache. I already tried using statics but that did not really help. I would like there to be only one instance per lsp.

<img width="621" alt="Screenshot 2024-02-06 at 12 03 45" src="https://github.com/phpactor/phpactor/assets/866743/1abc3395-f207-4e7f-a771-94b5fb6aad42">

<img width="688" alt="Screenshot 2024-02-06 at 12 05 05" src="https://github.com/phpactor/phpactor/assets/866743/72f7f539-2a47-4d46-b967-80e81287053d">

**Comment by dantleech** [comment]
You had me worried. I think what you are seeing is the diagnostics worker process which gets spawned pretty much on every keystroke. It's stateless and async. Set `"language_server.diagnostic_outsource": false,` and that behavior should disapear (but diagnostics will be calculated in the main process, slowing stuff down).

**Reply by haringsrob** [reply]
I will test that out!

**Reply by haringsrob** [reply]
Yea that indeed solves that!
